utils/use_websocket/hooks.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useContext, useEffect} from 'react';
+
+import {WebSocketClient, WebSocketMessage} from '@mattermost/client';
+
+import {WebSocketContext} from './context';
+
+export type UseWebSocketOptions = {
+    handler: (msg: WebSocketMessage) => void;
+}
+
+export function useWebSocket({handler}: UseWebSocketOptions) {
+    const wsClient = useWebSocketClient();
+
+    useEffect(() => {
+        wsClient.addMessageListener(handler);
+
+        return () => {
+            wsClient.removeMessageListener(handler);
+        };
+    }, [wsClient, handler]);
```

Comment:
Do we need these? Or can we just do `[]` so if someone doesn't remember `useCallback` on the handler it won't go nuts.

Reply:
We need that or else the `useWebsocket` will only ever register the first version of the handler that's provided with whatever is captured by it. If the handler depends on props or anything else, it'll continue to use the first value that was passed in.

Even if someone forgets to use `useCallback` though, I don't think this is that big of a deal though. Adding and removing these handlers is just modifying a Set, so there shouldn't be any noticeable performance hit by doing that on every render cycle. If it does become an issue though, we can always look at making a custom ESLint rule to error out when that happens, similar to `react-hooks/exhaustive-deps` which we have for `useEffect` already